nevergrad/optimization/optimizerlib.py

```diff
@@ -1522,35 +1531,63 @@ def _internal_provide_recommendation(self) -> ArrayLike:
             return self.current_center
 
     def _internal_ask_candidate(self) -> p.Parameter:
-        individual = self.current_center + self.sigma * self._rng.normal(0, 1, self.dimension)
+        tmp = self.sigma
+        if self.populationSizeAdapatation and self.popsize.llambda < self.paraLimit * self.dimension: # Population size not large enough for emna
+            mutated_sigma = self.sigma * np.exp(self._rng.normal(0, 1) / np.sqrt(self.dimension))
+            individual = self.current_center + mutated_sigma * self._rng.normal(0, 1, self.dimension)
+            tmp = mutated_sigma
+        else:
+            individual = self.current_center + self.sigma * self._rng.normal(0, 1, self.dimension)
         parent = self.parents[self.num_ask % len(self.parents)]
         candidate = parent.spawn_child().set_standardized_data(individual, reference=self.parametrization)
         if parent is self.parametrization:
             candidate.heritage["lineage"] = candidate.uid
-        candidate._meta["sigma"] = self.sigma
+        candidate._meta["sigma"] = tmp
         return candidate
 
     def _internal_tell_candidate(self, candidate: p.Parameter, value: float) -> None:
         candidate._meta["loss"] = value
+        if self.populationSizeAdapatation:
```

Comment:
to simplify the code you can probably always use the mu and llambda from the popsize instance, and either update it with add_value, or manually by directly overriding llambda and mu. This way there would be a unique source of truth for llambda and mu